src/core/Akka.TestKit/TestKitBase_ExpectAllWithPredicates.cs

```diff
@@ -0,0 +1,155 @@
+        await foreach (var item in enumerable)
+        {
+            yield return item;
+        }
+    }
+
+    private async IAsyncEnumerable<object> InternalExpectMsgAllOfMatchingPredicatesAsync(
+        TimeSpan max,
+        IReadOnlyCollection<PredicateInfo> predicateInfos,
+        bool shouldLog = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var predicateInfoList = predicateInfos.ToList();
+
+        ConditionalLog(shouldLog, "Expecting {0} messages during {1}", predicateInfos.Count, max);
+        var start = Now;
+
+        var unexpectedMessages = new List<object>();
+        var receivedMessages = InternalReceiveNAsync(predicateInfos.Count, max, shouldLog, cancellationToken)
+            .ConfigureAwait(false).WithCancellation(cancellationToken);
+        await foreach (var msg in receivedMessages)
+        {
+            var foundPredicateInfo = predicateInfoList.FirstOrDefault(p =>
+                p.Type == msg.GetType() && (bool)p.PredicateT.DynamicInvoke(msg));
+            if (foundPredicateInfo == null)
+                unexpectedMessages.Add(msg);
+            else
+            {
+                predicateInfoList.Remove(foundPredicateInfo);
+            }
+
+            yield return msg;
+        }
+
+        CheckMissingAndUnexpected(predicateInfoList, unexpectedMessages, "not found", "found unexpected", shouldLog,
+            $"Expected {predicateInfos.Count} messages during {max}. Failed after {Now - start}. ");
+    }
+}
+
+public class PredicateInfo
+{
+    public Type Type { get; }
+    public Delegate PredicateT { get; }
+
+    private PredicateInfo(Delegate predicateT, Type type)
+    {
+        Type = type;
+        PredicateT = predicateT;
+    }
+
+    public static PredicateInfo Create<T>(Predicate<T> predicateT)
+    {
+        return new PredicateInfo(predicateT, typeof(T));
+    }
+
+    public override string ToString()
+    {
+        return $"{GetType().FullName}<{Type.FullName}>";
+    }
+}
+
+
+
+public static class PredicateInfoFactory
+{
+    public static PredicateInfo CreatePredicateInfo<T>(this Predicate<T> predicateT)
+    {
+        return PredicateInfo.Create(predicateT);
+    }
+
+    public static PredicateInfo CreatePredicateInfo<T>(this Delegate predicateT)
+    {
+        var error = "";
+
+        var method = predicateT.Method;
+        if (method.ReturnType != typeof(bool))
+        {
+            error = "Return type of this Delegate is not a boolean. ";
+        }
+
+        var parameters = method.GetParameters();
+        if (parameters.Length != 1)
+        {
+            error += "Delegate does not take exactly one parameter.";
+        }
+
+        if (error.Length != 0)
+            throw new ArgumentException(error);
+
+        var predicate =
+            (Predicate<int>)Delegate.CreateDelegate(typeof(Predicate<int>), predicateT.Target, predicateT.Method);
```

Comment:
Shouldn't this be `Predicate<T>` instead of `Predicate<int>`?

Reply:
You are correct. Copy/paste error.